index.d.ts

```diff
@@ -59,9 +100,9 @@ export interface AnyAction extends Action {
  * @template A The type of actions the reducer can potentially respond to.
  */
 export type Reducer<S = any, A extends Action = AnyAction> = (
-  state: S | undefined,
+  state: UnCombinedState<S> | undefined,
   action: A
-) => S
+) => UnCombinedState<S>
```

Comment:
Hmm, this is the only part I really have issue with, mainly from an ergonomics perspective. The type returned from the reducer is now this wrapped state type, not my original state. I know it is evaluated to be that original state, but won't it appear in TS error traces? That seems a bit confusing, especially for a central type.

Reply:
I'd almost suggest renaming "UnCombinedState" to simply "State", which both reflects that this is the main state and hopefully be less confusing naming in error traces. Otherwise I'm delighted to see that there is now a way to type the PreloadedState properly.

Reply:
How about `ReducerState` since it is the state as normalized for input to and output from a `Reducer`?

Reply:
Never mind. I found a way to make the `[$CombinedState]` property optional, which means it doesn't need to be stripped out in `Reducer`, and `UnCombinedState` is no longer necessary at all.